server/middleware/featureGate.test.ts

```diff
@@ -1,43 +1,95 @@
 import type { NextFunction, Request, Response, RequestHandler } from 'express'
-import featureGate from './featureGate'
+
+import { featureGate, caseloadGate } from './featureGate'
+
+/** Trivial request handler that always returns "OK" */
+const simpleRequestHandler: RequestHandler = (req, res) => {
+  res.send('OK')
+}
+
+/**
+ * Creates a mock request with given app locals
+ */
+function mockRequest(appLocals: Record<string, unknown> = {}): Request {
+  return {
+    app: { locals: appLocals },
+  } as Request
+}
+
+/**
+ * Creates a mock response with given response locals
+ * NB: the send method will track calls
+ */
+function mockResponse(locals: Record<string, unknown> = {}): Response {
+  return {
+    locals,
+    send: jest.fn(),
+  } as unknown as Response
```

Comment:
how does this `as unknown as Response` work?

Reply:
`Response` is a huge and complicated type. the internet suggests that you mock and set the bits you need. `as Response` does not work because the type of this partial object is not compatible. so you need to force the type assertion. anything can be asserted as being `unknown` and anything can be made out of an `unknown` with a type assertion.

basically you gotta do this unless you want to implement 40 properties on this object 🤷

Reply:
Oh I see now, you trick TS to think that's is `unknown` and then trick it again to think it's a `Response`: It's still missing many properties but now TS doesn't know 👌